### PR TITLE
Unit tests for transfer, vote and transaction logic

### PR DESCRIPTION
**Pull request by SargeKhan**

### CODE REVIEW

File: test/unit/logic/transaction.js [hunk 1]
```diff
@@ -0,0 +1,1076 @@
...
+	before(function (done) {
+		async.auto({
+			rounds: function (cb) {
+				modulesLoader.initModule(Rounds, modulesLoader.scope,cb);
+			},
+			accountLogic: function (cb) {
+				modulesLoader.initLogicWithDb(AccountLogic, cb);
+			},
+			transaction: ['accountLogic', function (result, cb) {
+				modulesLoader.initLogicWithDb(Transaction, cb, {
```
Comment (4miners) [inline test/unit/logic/transaction.js]: Are you sure that `Transaction` logic is initialised properly? It require also `schema`, `genesisblock`, `logger`.

Comment (SargeKhan) [inline test/unit/logic/transaction.js]: This object which I pass gets merged default `scope` in initModule here: https://github.com/SargeKhan/lisk/blob/6037bdc5d3daf59eb4f5eeba1d864660f21e49b9/test/common/initModule.js#L208

Comment (4miners) [inline test/unit/logic/transaction.js]: Ok.

**Comment by 4miners** [comment]
@SargeKhan Great work 👍

**Pull request merged**